pymc3/tests/test_sampling.py

```diff
@@ -936,14 +933,14 @@ def test_ignores_observed(self):
         npt.assert_array_almost_equal(prior["positive_mu"], np.abs(prior["mu"]), decimal=4)
 
     def test_respects_shape(self):
-        for shape in (2, (2,), (10, 2), (10, 10)):
+        for shape in ((2,), (10, 2), (10, 10)):
             with pm.Model():
-                mu = pm.Gamma("mu", 3, 1, size=1)
-                goals = pm.Poisson("goals", mu, size=shape)
+                mu = pm.Gamma("mu", 3, 1)
```

Comment:
You've removed `size=1`, but are you sure that this wasn't supposed to be part of the test?

Also, why remove it in the first place?

Reply:
Making `mu` a length-1 vector and also passing `size=shape` in `Poisson` makes the resulting RV `tuple(shape + (1,))`.
Consequently the assert of trace shape fails.

Again, this test only worked before, because the effect of `size` were inconsistent.

Reply:
After our misunderstanding I'm now revisiting this test to evaluate if it should be reverted.
So what the test did before my changes:
1. Create a `(1,)` `Gamma` RV
2. Use that as the `mu` parameter in a `Poisson`, but then set the `size` to `(2,)`, `10, 2)` and so on,

So essentially it did not respect that `mu` was actually a length 1 _vector_ and treated it as if it was a scalar.
Was it really the intention of the test to verify that a lengh-1 vector input parameter is treated like a scalar, or is that just a leftover from our old shape handling before #4214, when `size=1` became scalar?